scripts/buildtable.pl

```diff
@@ -96,19 +96,28 @@
 my $bipnum = 0;
 while (++$bipnum <= $topbip) {
 	my $fn = sprintf "bip-%04d.mediawiki", $bipnum;
+	my $is_markdown = 0;
 	if (!-e $fn) {
 		$fn = sprintf "bip-%04d.md", $bipnum;
+		$is_markdown = 1;
 	}
 	-e $fn || next;
 	open my $F, "<$fn";
-	while (<$F> !~ m[^(?:\xef\xbb\xbf)?<pre>$]) {
+	if ($is_markdown) {
+		while (<$F> !~ m[^(?:\xef\xbb\xbf)?```$]) {
+			die "No ``` in $fn" if eof $F;
+		}
+	} else {
+		while (<$F> !~ m[^(?:\xef\xbb\xbf)?<pre>$]) {
```

Comment:
This appears to be Perl for reading a file line by line, and requiring that the initial lines of the file either only contain the byte order mark, or <code>```</code> in the first line if it is a Markdown file or `<pre>` if it is a Mediawiki file.

Reply:
It reads the next (first) line of the file, checks if the full line doesn't exactly match `<pre>` possibly preceded by the byte order marker, and aborts if it doesn't. Not sure why it's written as a while loop; it's not like it can actually enter the loop body twice.